app/main.development.js

```diff
@@ -55,12 +59,66 @@ if (argv.version) {
   app.exit(0);
 }
 
-// Check if network was set on command line (but only allow one!).
+let rpcOptionsCount = 0;
+rpcOptionsCount += argv.rpcuser ? 1 : 0;
+rpcOptionsCount += argv.rpcpass ? 1 : 0;
+rpcOptionsCount += argv.rpccert ? 1 : 0;
+rpcOptionsCount += argv.rpcconnect ? 1 : 0;
+
+// Allow at most one network to be specified
 if (argv.testnet && argv.mainnet) {
-  logger.log(BOTH_CONNECTION_ERR_MESSAGE);
+  console.log(BOTH_CONNECTION_ERR_MESSAGE);
+  app.quit();
+} else if (!argv.spv && argv.spvconnect !== undefined) {
+  console.log(SPV_CONNECT_WITHOUT_SPV);
+  app.quit();
+} else if (argv.spv && argv.advanced) {
+  console.log(SPV_WITH_ADVANCED_MODE);
+  app.quit();
+} else if (!argv.advanced && (argv.rpcuser || argv.rpcpass || argv.rpccert || argv.rpcconnect)) {
+  console.log(RPC_WITHOUT_ADVANCED_MODE);
+  app.quit();
+} else if (rpcOptionsCount > 0 && rpcOptionsCount < 4) {
+  console.log(RPC_MISSING_OPTIONS);
   app.quit();
 }
 
+// Signal to renderer process that CLI options should override the global config
+if (argv.testnet) {
+  cliOptions.network = "testnet";
+} else if (argv.mainnet) {
+  cliOptions.network = "mainnet";
+}
+if (argv.advanced) {
+  cliOptions.daemonStartAdvanced = true;
+}
+if (argv.spv) {
+  cliOptions.spvMode = true;
+}
+if (argv.spvconnect !== undefined && isString(argv.spvconnect)) {
+  cliOptions.spvConnect = argv.spvconnect.split(",");
+}
+if (argv.rpcuser !== undefined && isString(argv.rpcuser)) {
+  cliOptions.rpcUser = argv.rpcuser;
+}
+if (argv.rpcpass !== undefined && isString(argv.rpcpass)) {
+  cliOptions.rpcPass = argv.rpcpass;
+}
+if (argv.rpccert !== undefined && isString(argv.rpccert)) {
+  cliOptions.rpcCert = argv.rpccert;
+}
+if (argv.rpcconnect !== undefined && isString(argv.rpcconnect)) {
+  const parts = argv.rpcconnect.split(":");
+  // Allowed formats: "127.0.0.1" or "127.0.0.1:19109"
+  if (parts.length !== 1 && parts.length !== 2) {
+    logger.log("error", RPCCONNECT_INVALID_FORMAT);
+    app.quit();
+  }
+  cliOptions.rpcHost = parts[0];
+  cliOptions.rpcPort = parts[1];
+}
+cliOptions.rpcPresent = rpcOptionsCount > 0 ? true : false;
```

Comment:
Maybe this should be `rpcOptionsCount >= 4` instead? So that rpcPresent will only be true if _all_ options are set? Would that work?

Reply:
Due to the validation of RPC options above, this should never execute with `rpcOptionsCount` having a value other than 0 or 4, so in terms of outside behavior the two checks should be identical. I'm fine with it though as it might be more self-explanatory for future devs reading the code. Better yet, `==` could be even better for signaling intent.